File: src/listeners/Ready.ts
import { Listener } from "@newtykins/botkit";
import { ActivityType } from "discord.js";

@Listener.Config({
	name: 'ready',
	event: Listener.Events.ClientReady,
	once: true
})
export default class Ready extends Listener<typeof Listener.Events.ClientReady> {
	async run() {
		this.logger.info(`Logged in as ${this.client.user?.tag}!`);

		this.client.user?.setPresence({
			activities: [
				{
					name: ':3',
					type: ActivityType.Streaming,
					url: 'https://www.twitch.tv/monstercat'
				}
			]
		})
	}
}

File: src/events/ready.rs
use johnny::preludes::event::*;
use owo_colors::OwoColorize;
#[cfg(johnny)]
use poise::serenity_prelude::Activity;
use poise::serenity_prelude::Ready;

pub async fn run(
    #[cfg(any(johnny, sqlite))] ctx: &Context,
    ready: &Ready,
    data: &Data,
) -> Result<()> {
    data.logger
        .info(format!("Logged in as {}", ready.user.name.bold()), None)
        .await?;

    // set the activity
    #[cfg(johnny)]
    ctx.set_activity(Activity::streaming(":3", "https://twitch.tv/monstercat"))
        .await;

    // ? is 100 really a sane quantity?
    #[cfg(sqlite)]
    if ctx.cache.guild_count() > 100 {
        data.logger
            .warn(
                format!(
                    "Your server is in {} guilds. Consider migrating from sqlite to either postgres or mysql.",
                    ctx.cache.guild_count()
                ),
                None,
            )
            .await?;
    }

    Ok(())
}
